### PR TITLE
Create a Food101 Model

### LINKED ISSUE #7: Develop ML Model for Food101 Kaggle Dataset
Create simple ml model for image recognition using the Food101 Kaggle dataset.

Document setup and explain code concisely.

### PR DESCRIPTION
**Pull request by GAchuzia**
# Description

Please include a summary of the changes and the related issue.

- Created a simple food classification model based on ResNet trained on the ETH Zurich Food 101 dataset.
- Includes setup required for downloading dataset and checking if GPU is available for training
- Trained model for 50 epochs, best model has training accuracy of 99.82% and validation accuracy of 85.91%

image

- Lastly, a quick test for the discord PR Bot webhook

Fixes #7 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix/feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

- [ ] Unit tests
- [x] Manual tests
- [ ] Linted/formatted

## Checklist

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Indecisive613 — CHANGES_REQUESTED** [review]
There is something non-platform wide about the imports you added:
image

Could you edit these lines to make them universal to all systems? You could choose to have a universal line, or mandate a specific version for us and document your decision. For reference, I have CUDA version 12.7.

**Review by Indecisive613 — APPROVED** [review]
Compiles properly and I can run setup.py. I did not test train.py due to timing constraints, but I trust that it is good.
